src/transformers/models/blip_2/modeling_blip_2.py

```diff
@@ -1448,19 +1449,21 @@ class Blip2Model(Blip2PreTrainedModel):
     def __init__(self, config: Blip2Config):
         super().__init__(config)
 
-        self.vision_model = Blip2VisionModel(config.vision_config)
+        self.vision_model = Blip2VisionModel._from_config(
+            config.vision_config, attn_implementation=config._attn_implementation["vision_config"]
+        )
```

Comment:
my main question is why is this info not stored in the config.vision_config directly? 
Initializing the composite config should propagate the `attn_implementation` attribute to the sub configs, instead of extracting it in modeling

Reply:
That means updating `from_config` to support extracting the attn_implementation from the config 😉

Reply:
oh right! The was the initial plan was similar, but it was supposed to recursively call `autoset_attn_impl` from within `autoset_attn_impl` for each sub-module-class. But that was too much extra code and checks

But I think I got you mean here, like instead of doing a dict for generatl config we can simply propagare user-requested attn for each sub-config. Then we hope it will get picked up in later `AutoModel.from_config` calls. Good idea and a lot cleaner, let me see how it goes. Hopefully that way we don't need specific treatment for nested configs

Reply:
🤗 good luck!